tower-request-modifier/tests/request_modification.rs

```diff
@@ -0,0 +1,136 @@
+extern crate http;
+extern crate tower_mock;
+extern crate tower_request_modifier;
+extern crate tower_service;
+
+use http::uri::{Authority, Scheme};
+use http::{Request, Response};
+use tower_mock::*;
+use tower_request_modifier::BuilderError;
+use tower_request_modifier::{Builder, RequestModifier};
+use tower_service::Service;
+
+#[test]
+fn adds_origin_to_requests() {
+    let scheme = Scheme::HTTP;
+    let authority: Authority = "www.example.com".parse().unwrap();
+
+    let (mock, mut handle) = Mock::new();
+
+    let mut add_origin = Builder::new()
+        .set_origin("http://www.example.com")
+        .build(mock)
+        .unwrap();
+
+    let request = Request::get("/").body(()).unwrap();
+
+    let _response = add_origin.call(request);
+
+    // Get the request
+    let request = handle.next_request().unwrap();
+    let (request, send_response) = request.into_parts();
+
+    // Assert that the origin is set
+    assert_eq!(request.uri().scheme_part().unwrap(), &scheme);
+    assert_eq!(request.uri().authority_part().unwrap(), &authority);
+
+    // Make everything happy:
+    let response = Response::builder().status(204).body(());
+
+    send_response.respond(response);
+}
+
+#[test]
+fn adds_header_to_requests() {
+    let header = "authorization";
+    let token = "Bearer ee2c2e06-0254-441d-b885-5bade6d7f3b2";
+
+    let (mock, mut handle) = Mock::new();
+
+    let mut add_token = Builder::new()
+        .add_header(header, token)
+        .build(mock)
+        .unwrap();
+
+    let request = Request::get("/").body(()).unwrap();
+
+    let _response = add_token.call(request);
+
+    // Get the request
+    let (request, send_response) = handle.next_request().unwrap().into_parts();
+
+    // Assert that the token header is set
+    assert!(request.headers().contains_key(header.to_owned()));
+
+    // Make everything happy:
+    let response = Response::builder().status(204).body(());
+
+    send_response.respond(response);
+}
+
+#[test]
+fn run_arbitrary_modifier() {
+    let (mock, mut handle) = Mock::new();
```

Comment:
(Mobile, sorry!) Tower just switched to Tower-test, with mock being exported. Mock::new’s been replaced by Mock::pair (with some friendlier  method names, too). Consider delete your cargo.lock and pulling in those (breaking) changes?

Reply:
(sorry)

Reply:
Ah, thank you!